wicket-core/src/main/java/org/apache/wicket/core/util/string/JavaScriptUtils.java

```diff
@@ -152,28 +155,48 @@ public static void writeJavaScriptUrl(final Response response, final CharSequenc
 	public static void writeJavaScriptUrl(final Response response, final CharSequence url,
```

Comment:
These method have to much arguments already: Just give it an IValueMap and write all out to the tag.

Reply:
I didn't quite get you.  Do you want the `writeJavaScriptUrl(final Response response, final CharSequence url, final String id, boolean defer, String charset, boolean async)` to be dropped?

Reply:
Well, if we continue what we have done in the past, we have to add yet another parameter "nonce".  IMHO it's better to change it to receive a valueMap instead, and let the caller fill it.

Reply:
There's a new method just after this one. I suppose we can mark this one as Deprecated and get rid of all usages in the wicket code.

Reply:
Ah yes, that's what I was looking for :). This one can be deprecated and should no longer be called.